### ISSUE TITLE
resample: points in dimension 0 must be strictly ascending

### ISSUE THREAD

**Issue by stscieisenhamer** [spacetelescope/jwst#2395]
# Issue
In working with recent SDP data on another issue, #2338, in trying to reproduce with current master gives ultimately the following error. The data has been placed in
```
/grp/jwst/ssb/eisenhamer/i2338
```
In hopes the error could have been mitigated, a full run from `detector1` on up is documented in the notebook `process.ipynb`. Note that this is a new error since milestone 0.10.

Full log:
```
2018-08-20 15:42:41,786 - stpipe.Spec3Pipeline - INFO - Spec3Pipeline instance created.
2018-08-20 15:42:41,788 - stpipe.Spec3Pipeline.mrs_imatch - INFO - MRSIMatchStep instance created.
2018-08-20 15:42:41,790 - stpipe.Spec3Pipeline.outlier_detection - INFO - OutlierDetectionStep instance created.
2018-08-20 15:42:41,791 - stpipe.Spec3Pipeline.resample_spec - INFO - ResampleSpecStep instance created.
2018-08-20 15:42:41,794 - stpipe.Spec3Pipeline.cube_build - INFO - CubeBuildStep instance created.
2018-08-20 15:42:41,796 - stpipe.Spec3Pipeline.extract_1d - INFO - Extract1dStep instance created.
2018-08-20 15:42:41,797 - stpipe - INFO - Hostname: inanna2.stsci.edu
2018-08-20 15:42:41,798 - stpipe - INFO - OS: Darwin
2018-08-20 15:42:41,843 - stpipe.Spec3Pipeline - INFO - Step Spec3Pipeline running with args ('output/jw98010-o001_20180806t184118_spec3_001_asn.json',).
2018-08-20 15:42:42,051 - stpipe.Spec3Pipeline - INFO - Prefetching reference files for dataset: 'jw98010001001_02104_00001_mirimage_cal.fits' reftypes = ['cubepar', 'drizpars', 'extract1d', 'resol']
2018-08-20 15:42:42,055 - stpipe.Spec3Pipeline - INFO - Prefetch for CUBEPAR reference file is 'N/A'.
2018-08-20 15:42:42,056 - stpipe.Spec3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/miri/jwst_miri_drizpars_0001.fits'.
2018-08-20 15:42:42,057 - stpipe.Spec3Pipeline - INFO - Prefetch for EXTRACT1D reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/miri/jwst_miri_extract1d_0004.json'.
2018-08-20 15:42:42,058 - stpipe.Spec3Pipeline - INFO - Prefetch for RESOL reference file is 'N/A'.
2018-08-20 15:42:42,243 - stpipe.Spec3Pipeline - INFO - Prefetching reference files for dataset: 'jw98010001001_02103_00001_mirimage_cal.fits' reftypes = ['cubepar', 'drizpars', 'extract1d', 'resol']
2018-08-20 15:42:42,247 - stpipe.Spec3Pipeline - INFO - Prefetch for CUBEPAR reference file is 'N/A'.
2018-08-20 15:42:42,248 - stpipe.Spec3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/miri/jwst_miri_drizpars_0001.fits'.
2018-08-20 15:42:42,249 - stpipe.Spec3Pipeline - INFO - Prefetch for EXTRACT1D reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/miri/jwst_miri_extract1d_0004.json'.
2018-08-20 15:42:42,251 - stpipe.Spec3Pipeline - INFO - Prefetch for RESOL reference file is 'N/A'.
2018-08-20 15:42:42,486 - stpipe.Spec3Pipeline - INFO - Prefetching reference files for dataset: 'jw98010001001_02101_00001_mirimage_cal.fits' reftypes = ['cubepar', 'drizpars', 'extract1d', 'resol']
2018-08-20 15:42:42,490 - stpipe.Spec3Pipeline - INFO - Prefetch for CUBEPAR reference file is 'N/A'.
2018-08-20 15:42:42,491 - stpipe.Spec3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/miri/jwst_miri_drizpars_0001.fits'.
2018-08-20 15:42:42,492 - stpipe.Spec3Pipeline - INFO - Prefetch for EXTRACT1D reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/miri/jwst_miri_extract1d_0004.json'.
2018-08-20 15:42:42,493 - stpipe.Spec3Pipeline - INFO - Prefetch for RESOL reference file is 'N/A'.
2018-08-20 15:42:42,745 - stpipe.Spec3Pipeline - INFO - Prefetching reference files for dataset: 'jw98010001001_02102_00001_mirimage_cal.fits' reftypes = ['cubepar', 'drizpars', 'extract1d', 'resol']
2018-08-20 15:42:42,749 - stpipe.Spec3Pipeline - INFO - Prefetch for CUBEPAR reference file is 'N/A'.
2018-08-20 15:42:42,750 - stpipe.Spec3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/miri/jwst_miri_drizpars_0001.fits'.
2018-08-20 15:42:42,750 - stpipe.Spec3Pipeline - INFO - Prefetch for EXTRACT1D reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/miri/jwst_miri_extract1d_0004.json'.
2018-08-20 15:42:42,752 - stpipe.Spec3Pipeline - INFO - Prefetch for RESOL reference file is 'N/A'.
2018-08-20 15:42:42,752 - stpipe.Spec3Pipeline - INFO - Starting calwebb_spec3 ...
2018-08-20 15:42:43,044 - stpipe.Spec3Pipeline.outlier_detection - INFO - Step outlier_detection running with args (<ModelContainer>,).
2018-08-20 15:42:43,047 - stpipe.Spec3Pipeline.outlier_detection - INFO - Performing outlier detection on 4 inputs
2018-08-20 15:42:43,092 - stpipe.Spec3Pipeline.outlier_detection - WARNING - /Users/eisenham/anaconda3/envs/jwstdev/lib/python3.6/site-packages/numpy/lib/function_base.py:4033: RuntimeWarning: All-NaN slice encountered
  r = func(a, **kwargs)

2018-08-20 15:42:43,874 - stpipe.Spec3Pipeline.outlier_detection - INFO - Resampling slit (387, 43)
----------------------------------------------------------------------
ERROR RUNNING STEP 'Spec3Pipeline':
    The points in dimension 0 must be strictly ascending
Traceback (most recent call last):
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/pipeline/calwebb_spec3.py", line 120, in process
    result = self.outlier_detection(result)
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/outlier_detection/outlier_detection_step.py", line 166, in process
    step.do_detection()
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/outlier_detection/outlier_detection_spec.py", line 75, in do_detection
    sdriz.do_drizzle()
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/resample/resample_spec.py", line 242, in do_drizzle
    pscale_ratio=self.pscale_ratio)
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/resample/gwcs_drizzle.py", line 205, in add_image
    fillval=self.fillval)
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/resample/gwcs_drizzle.py", line 425, in dodrizzle
    pixmap = resample_utils.calc_gwcs_pixmap(input_wcs, output_wcs, insci.shape)
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/resample/resample_utils.py", line 118, in calc_gwcs_pixmap
    pixmap = np.dstack(reproject(in_wcs, out_wcs)(grid[0], grid[1]))
  File "/Users/eisenham/Documents/ssbdev/jwst/jwst/resample/resample_utils.py", line 166, in _reproject
    det = backward_transform(*tuple(flat_sky))
  File "/Users/eisenham/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py", line 382, in __call__
    ('equivalencies', None)])
  File "/Users/eisenham/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py", line 363, in __call__
    return super(cls, self).__call__(*inputs, **kwargs)
  File "/Users/eisenham/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py", line 815, in __call__
    outputs = self.evaluate(*chain(inputs, parameters))
  File "/Users/eisenham/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py", line 2921, in evaluate
    return self.__class__.evaluate(*args)
  File "/Users/eisenham/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py", line 2362, in evaluate
    result = cls._evaluate(inputs, params)
  File "/Users/eisenham/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py", line 2194, in <lambda>
    return (lambda inputs, params: g[0](f[0](inputs, params), params),
  File "/Users/eisenham/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py", line 2206, in <lambda>
    g[0](inputs[f[1]:], params)),
  File "/Users/eisenham/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py", line 2802, in f
    return evaluate_wrapper(model, inputs, param_values)
  File "/Users/eisenham/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py", line 2794, in evaluate_wrapper
    outputs = model.evaluate(*inputs, *param_values)
  File "/Users/eisenham/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/tabular.py", line 225, in evaluate
    fill_value=self.fill_value)
  File "/Users/eisenham/anaconda3/envs/jwstdev/lib/python3.6/site-packages/scipy/interpolate/interpolate.py", line 2621, in interpn
    "ascending" % i)
ValueError: The points in dimension 0 must be strictly ascending

----------------------------------------------------------------------
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-18-16d7bad5aa6f> in <module>()
      5     'output'
      6 ]
----> 7 Step.from_cmdline(args)

~/Documents/ssbdev/jwst/jwst/stpipe/step.py in from_cmdline(args)
    160         """
    161         from . import cmdline
--> 162         return cmdline.step_from_cmdline(args)
    163 
    164     @classmethod

~/Documents/ssbdev/jwst/jwst/stpipe/cmdline.py in step_from_cmdline(args, cls)
    295             cProfile.runctx("step.run(*positional)", globals(), locals(), profile_path)
    296         else:
--> 297             step.run(*positional)
    298     except Exception as e:
    299         import traceback

~/Documents/ssbdev/jwst/jwst/stpipe/step.py in run(self, *args)
    398             else:
    399                 try:
--> 400                     step_result = self.process(*args)
    401                 except TypeError as e:
    402                     if "process() takes exactly" in str(e):

~/Documents/ssbdev/jwst/jwst/pipeline/calwebb_spec3.py in process(self, input)
    118 
    119             # Call outlier detection
--> 120             result = self.outlier_detection(result)
    121 
    122             # Resample time. Dependent on whether the data is IFU or

~/Documents/ssbdev/jwst/jwst/stpipe/step.py in run(self, *args)
    398             else:
    399                 try:
--> 400                     step_result = self.process(*args)
    401                 except TypeError as e:
    402                     if "process() takes exactly" in str(e):

~/Documents/ssbdev/jwst/jwst/outlier_detection/outlier_detection_step.py in process(self, input)
    164             # Set up outlier detection, then do detection
    165             step = detection_step(self.input_models, reffiles=reffiles, **pars)
--> 166             step.do_detection()
    167 
    168             if self.input_container:

~/Documents/ssbdev/jwst/jwst/outlier_detection/outlier_detection_spec.py in do_detection(self)
     73                                                    single=True,
     74                                                    **pars)
---> 75             sdriz.do_drizzle()
     76             drizzled_models = sdriz.output_models
     77             for model in drizzled_models:

~/Documents/ssbdev/jwst/jwst/resample/resample_spec.py in do_drizzle(self, **pars)
    240                 driz.add_image(img.data, in_wcs, inwht=inwht,
    241                         expin=img.meta.exposure.exposure_time,
--> 242                         pscale_ratio=self.pscale_ratio)
    243 
    244             # Update some basic exposure time values based on all the inputs

~/Documents/ssbdev/jwst/jwst/resample/gwcs_drizzle.py in add_image(self, insci, inwcs, inwht, xmin, xmax, ymin, ymax, pscale_ratio, expin, in_units, wt_scl)
    203                             xmin=xmin, xmax=xmax, ymin=ymin, ymax=ymax,
    204                             pixfrac=self.pixfrac, kernel=self.kernel,
--> 205                             fillval=self.fillval)
    206 
    207     def blot_image(self, blotwcs, interp='poly5', sinscl=1.0):

~/Documents/ssbdev/jwst/jwst/resample/gwcs_drizzle.py in dodrizzle(insci, input_wcs, inwht, output_wcs, outsci, outwht, outcon, expin, in_units, wt_scl, pscale_ratio, uniqid, xmin, xmax, ymin, ymax, pixfrac, kernel, fillval)
    423     # Compute the mapping between the input and output pixel coordinates
    424     # for use in drizzle.cdrizzle.tdriz
--> 425     pixmap = resample_utils.calc_gwcs_pixmap(input_wcs, output_wcs, insci.shape)
    426     # pixmap[np.isnan(pixmap)] = -10
    427     # print("Number of NaNs: ", len(np.isnan(pixmap)) / 2)

~/Documents/ssbdev/jwst/jwst/resample/resample_utils.py in calc_gwcs_pixmap(in_wcs, out_wcs, shape)
    116 
    117     grid = wcstools.grid_from_bounding_box(bb, step=(1, 1))
--> 118     pixmap = np.dstack(reproject(in_wcs, out_wcs)(grid[0], grid[1]))
    119     return pixmap
    120 

~/Documents/ssbdev/jwst/jwst/resample/resample_utils.py in _reproject(x, y)
    164         for axis in sky:
    165             flat_sky.append(axis.flatten())
--> 166         det = backward_transform(*tuple(flat_sky))
    167         det_reshaped = []
    168         for axis in det:

~/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py in __call__(self, x00, x10, x01, model_set_axis, with_bounding_box, fill_value, equivalencies)
    380                                      ('with_bounding_box', False),
    381                                      ('fill_value', np.nan),
--> 382                                      ('equivalencies', None)])
    383 
    384             # The following makes it look like __call__ was defined in the class

~/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py in __call__(self, *inputs, **kwargs)
    361             def __call__(self, *inputs, **kwargs):
    362                 """Evaluate this model on the supplied inputs."""
--> 363                 return super(cls, self).__call__(*inputs, **kwargs)
    364 
    365             # When called, models can take two optional keyword arguments:

~/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py in __call__(self, *inputs, **kwargs)
    813                     outputs = [np.asarray(r) for r in result]
    814         else:
--> 815             outputs = self.evaluate(*chain(inputs, parameters))
    816         if self.n_outputs == 1:
    817             outputs = (outputs,)

~/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py in evaluate(self, *args)
   2919     @sharedmethod
   2920     def evaluate(self, *args):
-> 2921         return self.__class__.evaluate(*args)
   2922 
   2923     # TODO: The way this works is highly inefficient--the inverse is created by

~/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py in evaluate(cls, *args)
   2360         inputs = args[:cls.n_inputs]
   2361         params = iter(args[cls.n_inputs:])
-> 2362         result = cls._evaluate(inputs, params)
   2363         if cls.n_outputs == 1:
   2364             return result[0]

~/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py in <lambda>(inputs, params)
   2192     #
   2193     # and similarly for g
-> 2194     return (lambda inputs, params: g[0](f[0](inputs, params), params),
   2195             f[1], g[2])
   2196 

~/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py in <lambda>(inputs, params)
   2204     # and similarly for g
   2205     return (lambda inputs, params: (f[0](inputs[:f[1]], params) +
-> 2206                                     g[0](inputs[f[1]:], params)),
   2207             f[1] + g[1], f[2] + g[2])
   2208 

~/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py in f(inputs, params)
   2800             def f(inputs, params):
   2801                 param_values = tuple(islice(params, n_params))
-> 2802                 return evaluate_wrapper(model, inputs, param_values)
   2803         else:
   2804             # Where previously model was a class, now make an instance

~/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/core.py in evaluate_wrapper(model, inputs, param_values)
   2792         def evaluate_wrapper(model, inputs, param_values):
   2793             inputs = model._validate_input_units(inputs)
-> 2794             outputs = model.evaluate(*inputs, *param_values)
   2795             if n_outputs == 1:
   2796                 outputs = (outputs,)

~/anaconda3/envs/jwstdev/lib/python3.6/site-packages/astropy/modeling/tabular.py in evaluate(self, *inputs)
    223         result = interpn(self.points, self.lookup_table, inputs,
    224                          method=self.method, bounds_error=self.bounds_error,
--> 225                          fill_value=self.fill_value)
    226 
    227         # return_units not respected when points has no units

~/anaconda3/envs/jwstdev/lib/python3.6/site-packages/scipy/interpolate/interpolate.py in interpn(points, values, xi, method, bounds_error, fill_value)
   2619         if not np.all(np.diff(p) > 0.):
   2620             raise ValueError("The points in dimension %d must be strictly "
-> 2621                              "ascending" % i)
   2622         if not np.asarray(p).ndim == 1:
   2623             raise ValueError("The points in dimension %d must be "

ValueError: The points in dimension 0 must be strictly ascending
```

**Comment by jdavies-st** [comment]
I'll take a look at this issue.  I'm pretty sure I know what is causing this.

**Comment by stscieisenhamer** [comment]
auto-resolved didn't. resolved by #2422.

**Comment by jdavies-st** [comment]
So #2422 fixed the bug found here.  But this dataset also shows there is a bug somewhere in `outlier_detection`.  If one skips `outlier_detection` for this dataset, the pipeline runs through to completion.  I will open a separate issue for the `outlier_detection` bug.
